types/react-datepicker/index.d.ts

```diff
@@ -1,7 +1,8 @@
-import * as Popper from "@popperjs/core";
+import { Middleware, Placement, UseFloatingOptions } from "@floating-ui/react";
 import { Locale } from "date-fns";
 import * as React from "react";
-import { Modifier, StrictModifierNames } from "react-popper";
+
+type PopperProps = Partial<Omit<UseFloatingOptions, "open" | "whileElementsMounted" | "placement" | "middleware">>;
 
 export interface CalendarContainerProps {
     className?: string | undefined;
```

Comment:
According to https://github.com/Hacker0x01/react-datepicker/pull/4393/files#diff-bdf7d74fe2b1de77e4844501d2f292338d6b334b41806ecc917361f5a7afcca8, you should remove `show PopperArrow` and `arrowProps` props from `CalendarContainerProps`. This change was included in v5.0.0.

Reply:
done in last commit